src/Makefile

```diff
@@ -220,6 +227,10 @@ distclean:
 language:
 	../utils/extract_strings.pl -update
 
+test: CFLAGS += -DTEST=1 -I./tests/framework -coverage -g -O0 -fPIC
+test: tests/AllTests.c $(TARGET)_test.$(EXEEXT)
+	@echo Execute tests
+	./$(TARGET)_test.$(EXEEXT)
 
```

Comment:
I'd like to be able to run coverage_report.sh automatically after all tests have run (Ideally we'd build all tests 1st and run them all as a 2nd step...it would make debug easier).  I'm not sure how to do that with this build system though (and I don't think we can run coverage_report on a per-target basis when using travis, so we'd need something different there)